sklearn/utils/estimator_checks.py

```diff
@@ -1191,6 +1192,56 @@ def check_estimators_pickle(name, estimator_orig):
         assert_allclose_dense_sparse(result[method], unpickled_result)
 
 
+def check_pairwise_estimator_tag(name, estimator_orig):
+    attributes_to_check = [('metric', 'precomputed'),
+                           ('affinity', 'precomputed'),
+                           ('kernel', 'precomputed')]
+
+    # Using iris as sample data
+    iris = load_iris()
+    X, y_ = iris.data, iris.target
+    distance_matrix = pairwise_distances(X)
+
+    for attribute, attribute_value in attributes_to_check:
+        # Check to see if attribute value is supported by estimator
+        if getattr(estimator_orig, attribute, None) is not None:
+            continue
+        try:
+            # Construct new object of estimator with desired attribute value
+            modified_estimator = estimator_orig.__class__(
+                                    **{
+                                        attribute: attribute_value
+                                      })
+            # Not all estimators validate parameters, so check fit()
+            modified_estimator.fit(X=distance_matrix, y=y_)
```

Comment:
We usually pass X and y as positional attributes (some estimators have `Y` not `y`, for instance)

Reply:
Resolved